### PR TITLE
Upgrade to Hibernate Search 5.8.0.Beta4 : SPI cleanup

### PR DESCRIPTION
**Pull request by Sanne**
https://issues.jboss.org/browse/ISPN-8061

### CODE REVIEW

File: bom/pom.xml [hunk 1]
```diff
@@ -69,7 +69,7 @@
       <version.groovy>2.4.8</version.groovy>
       <version.hibernate.javax.persistence>1.0.0.Final</version.hibernate.javax.persistence>
       <version.hibernate.core>5.2.7.Final</version.hibernate.core>
-      <version.hibernate.search>5.8.0.Beta3</version.hibernate.search>
+      <version.hibernate.search>5.8.0.Beta4</version.hibernate.search>
```
Comment (anistor) [inline bom/pom.xml]: also need to update licenses.xml to point to beta4

Comment (Sanne) [inline bom/pom.xml]: Thanks, I had no idea. Fixed and rebased.

Comment (Sanne) [inline bom/pom.xml]: BTW you guys don't like automating such things anymore? It's a property..

**Comment by anistor** [comment]
LGTM

File: bom/pom.xml (hunk 1 above)
Comment (wburns) [inline bom/pom.xml]: Unfortunately the info in maven is not correct for everything :( We still have to work out the best way to correct that in an automated way, but until then it has to be done manually.

**Comment by tristantarrant** [comment]
This does not build.
[ERROR] Failed to execute goal on project infinispan-directory-provider: Could not resolve dependencies for project org.infinispan:infinispan-directory-provider:jar:9.1.0-SNAPSHOT: The following artifacts could not be resolved: org.hibernate:hibernate-search-engine:jar:5.8.0.Beta4, org.hibernate:hibernate-search-engine:jar:tests:5.8.0.Beta4, org.hibernate:hibernate-search-orm:jar:tests:5.8.0.Beta4, org.hibernate:hibernate-search-orm:jar:5.8.0.Beta4: Failure to find org.hibernate:hibernate-search-engine:jar:5.8.0.Beta4 in https://repository.jboss.org/nexus/content/groups/public-jboss was cached in the local repository, resolution will not be reattempted until the update interval of jboss-public-repository-group has elapsed or updates are forced -> [Help 1]

**Comment by anistor** [comment]
Beta4 is missing fron nexus for whatever reason. I have built HSearch 5.8.0.Beta4 locally and tested against it. I did not run full test suite, but the ones I thought to be relevant worked well.

**Comment by Sanne** [comment]
Sorry folks, we forgot to check if Nexus had closed the staging repository. Should work now.

**Comment by anistor** [comment]
This is going in!

**Comment by anistor** [comment]
integrated in master. Thanks @Sanne !

**Comment by Sanne** [comment]
THANKS!

**Pull request closed without merging**